### PR TITLE
[ci][config/5] make ci pipelines configurable

### PR DESCRIPTION
**Pull request by can-anyscale**
Make CI pipelines configurable, they are different upstream.

Test:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
is this ready for review or still in draft?

**Review by aslonnie — COMMENTED** [review]
I think this PR makes the pipeline separation to be a more general, and high-level abstraction, but not sure if it is the common / right thing to do. or what this is really for.

**Comment by can-anyscale** [comment]
the current scope is only so that I can delete some tech debts for upstream; we should turn it off correctly on upstream, rather than it happens to turn off because of the hard coded list of buildkite pipeline id; no plan to make this work with other repositories yet

the pipeline id can be defined as both pr and branch yes; the application logic should handle that case gracefully (e.g. the state machine will write data to either or both pr and branch test db bucket depending on how the branch is configured)

**Comment by can-anyscale** [comment]
the separation of pr and branch is pretty unique for ray (at least for the state machine application) because of the permission differences between these pipelines; otherwise it doesn't need separations

File: ci/ray_ci/oss_config.yaml [hunk 1]
```diff
@@ -6,6 +6,13 @@ release_byod:
   aws_cr: 029272617770.dkr.ecr.us-west-2.amazonaws.com
   gcp_cr: us-west1-docker.pkg.dev/anyscale-oss-ci
   aws2gce_credentials: release/aws2gce_iam.json
+ci_pipeline:
+  pr:
+    - 0189942e-0876-4b8f-80a4-617f988ec59b  # premerge
+  branch:
+    - 0189e759-8c96-4302-b6b5-b4274406bf89  # postmerge
+    - 018e0f94-ccb6-45c2-b072-1e624fe9a404  # postmerge-macos
+    - 018af6d3-58e1-463f-90ec-d9aa4a4f57f1  # release
```
[the remark this answers is not in the document]
Comment (can-anyscale) [reply ci/ray_ci/oss_config.yaml:15]: change to premerge/postmerge is ok-ish yes if we're fine with the 'release' pipeline to be under postmerge

[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/ray_ci/oss_config.yaml:15]: > if we're fine with the 'release' pipeline to be under postmerge

seems fine with me. or maybe "dev/reviewed" or "uncommitted/committed" ? `release` under `branch` also feels a bit weird.

[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/ray_ci/oss_config.yaml:15]: `uncommitted/committed` is probably the most accurate among all those.

`postmerge` feels like "immediately after merge", or at least has some kind of that meaning.

[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/ray_ci/oss_config.yaml:15]: or maybe just `premerge` and `merged` ..

**Review by aslonnie — COMMENTED** [review]
oh.. names ...

File: ci/ray_ci/oss_config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/ray_ci/oss_config.yaml:15]: or `unmerged` and `merged`.

**Comment by can-anyscale** [comment]
let's go with `premerge` and `postmerge` then, to overall have less concepts

**Comment by can-anyscale** [comment]
changed the name to 'premerge' and 'postmerge'

**Review by aslonnie — APPROVED** [review]

**Pull request merged**